src/Azure/Provider.php

```diff
@@ -91,17 +94,55 @@ protected function mapUserToObject(array $user)
      */
     protected function getTokenFields($code)
     {
-        return array_merge(parent::getTokenFields($code), [
+        $fields = [
             'grant_type' => 'authorization_code',
-            'resource'   => 'https://graph.windows.net',
-        ]);
+            'client_id' => $this->clientId,
+            'client_secret' => $this->clientSecret,
+            'code' => $code,
+            'redirect_uri' => $this->redirectUrl,
+        ];
+
+        if ($this->usesPKCE()) {
+            $fields['code_verifier'] = $this->request->session()->pull('code_verifier');
+        }
+
+        return $fields;
     }
 
     /**
      * {@inheritdoc}
      */
     public static function additionalConfigKeys()
     {
-        return ['tenant'];
+        return ['tenant', 'scopes', 'logout_url'];
```

Comment:
Scopes should be set when building the request, not in the config of the provider, so could you remove this?
Update the docs if required.

Reply:
I removed it from here.  getScopes()  too. 
Switched to protected $scopes attr